app/forms/foreign/allowances/ForeignZeroEmissionCarAllowanceFormProvider.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.foreign.allowances
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class ForeignZeroEmissionCarAllowanceFormProvider @Inject() extends Mappings {
+
+  val minimum = 0;
+  val maximum = 100000000;
+  def apply(individualOrAgent: String): Form[BigDecimal] =
+    Form(
+      "foreignZeroEmissionCarAllowanceAmount" -> currency(
+        s"foreignZeroEmissionCarAllowance.error.required.${individualOrAgent}",
+        s"foreignZeroEmissionCarAllowance.error.nonNumeric.${individualOrAgent}",
```

Comment:
this message key needs to be zeroEmissionCarAllowance.error.twoDecimalPlaces. Also could you confirm with Reddy what the error message should be for error.twoDecimalPlaces and error.nonNumeric because the content doesn't match with what we have for the UK property